akka-docs-dev/rst/java/code/docs/http/javadsl/server/HeaderRequestValsExampleTest.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (C) 2009-2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+
+package docs.http.javadsl.server;
+
+import akka.http.javadsl.model.HttpRequest;
+import akka.http.javadsl.model.headers.Host;
+import akka.http.javadsl.model.headers.RawHeader;
+import akka.http.javadsl.server.RequestVal;
+import akka.http.javadsl.server.Route;
+import akka.http.javadsl.server.values.Headers;
+import akka.http.javadsl.testkit.JUnitRouteTest;
+import org.junit.Test;
+
+public class HeaderRequestValsExampleTest extends JUnitRouteTest {
+
+  @Test
+  public void testHeaderVals() {
+    //#by-class
+    // extract the entire header instance:
+    RequestVal<Host> host = Headers.byClass(Host.class).instance();
```

Comment:
Yes, does the same thing. I could pick a different header for the example I guess.
`Host` is the most prominent in all http requests so I picked that one.